src/tests/ftest/server/metadata_svc_ops.py

```diff
@@ -0,0 +1,139 @@
+
+        Returns:
+            test_time: List of time consumed per test loops of metadata workload test cycles.
+
+        """
+        test_time = []
+        try:
+            container = self.get_container(pool, create=False)
+            if container.daos:
+                container.daos.verbose = False
+            container.create()
+            self.log.info("Successfully created #%s container", cont_num)
+        except (DaosTestError, TestFail) as err:
+            self.fail(
+                "#({}.{}) container create failed. err={}".format(pool.label, cont_num, err))
+        for ind in range(test_loops):
+            start = time.time()
+            for _ in range(workload_cycles):
+                container.close()
+                container.open()
+            elapsed_time = time.time() - start
+            self.log.info("Completed container Metadata test-loop: %d, elapsed_time: %f",
+                          ind + 1, elapsed_time)
+            test_time.append(elapsed_time)
+        for ind in range(test_loops):
+            self.log.info("Test time of Metadata test-loop: %d,  %f",
+                          ind, test_time[ind])
+        container.close()
+        return test_time
+
+    def test_metadata_dup_rpc(self):
+        """JIRA ID: DAOS-15937 metadata duplicate rpc detection time consuming.
+
+        Test Steps:
+            1. Bring up DAOS server.
+            2. Create pool1 with specified property svc_ops_entry_age.
+            3. Create containers by ThreadManager.
+            4. Run specified metadata workload cycles in multiple test loops (N cycles per loop),
+               calculate average time per loop, for those loops executed after svc_ops_entry_age
+               time has passed.
+            5. Create pool2 with property svc_ops_enable:0.
+            6. Create containers by ThreadManager on pool2.
+            7. Repeat test step 4 on pool2.
+            8. Compare metadata workload test time with and without duplicate rpc detection.
+
+        :avocado: tags=all,full_regression
+        :avocado: tags=hw,medium,md_on_ssd
+        :avocado: tags=server,metadata
+        :avocado: tags=DuplicateRpcDetection,test_metadata_dup_rpc
+        """
+        self.dmg.server_set_logmasks("DEBUG", raise_exception=False)
+        cont_num = self.params.get("number_thread", '/run/metadata/*', default=1)
+        w_cycles = self.params.get("workload_test_cycles", '/run/metadata/*', default=5000)
+        t_loops = self.params.get("test_loops", '/run/metadata/*', default=10)
+        threshold_factor = self.params.get("threshold_factor", '/run/metadata/*', default=1.75)
+
+        self.log_step("Create pool with properties svc_ops_entry_age.")
+        pool1 = self.get_pool(dmg=self.dmg.copy())
+
+        self.log_step("Create containers by ThreadManager.")
+        container_manager = ThreadManager(
+            self.metadata_workload_test, self.get_remaining_time() - 30)
+        container_manager.add(
```

Comment:
Thanks for the comment.
When multiple threads specified (in testcase-2, 2, 4, or 8 threads), each thread will create it's own container and perform their individual metadata workload. 
Have updated.  It should be number_of_container (instead of cont_num).

Reply:
If the yaml file specifies number_thread: 2 (anything larger than 1), then this code will need to change so that it has a loop to call container_manager.add() calls so that all of the threads get created, is that right?

Reply:
I agree with @kccain here. If `num_of_cont` is increased, this still only runs a single thread because the code only calls `container_manager.add` one time

> script will generate different number of container-thread (2, 4 or 8) on the same pool (thread)

Where? What line(s) of code does this?

Reply:
Line #107 to #110 generate different number of container-thread (2, 4 or 8) on the same pool.

Reply:
```
        container_manager = ThreadManager(
            self.metadata_workload_test, self.get_remaining_time() - 30)
        container_manager.add(
            pool=self.pool, num_of_cont=num_of_cont, workload_cycles=w_cycles, test_loops=t_loops)
```


That is one thread, no matter what `num_of_cont` is